src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java

```diff
@@ -177,9 +176,10 @@ public static void setupCodeSyncRepo(Project project, String repoPath, String re
 
                     if (shouldSyncRepo) {
                         syncRepo(repoPath, repoName, branchName, project, codeSyncProgressIndicator, false);
+                        reposBeingSynced.remove(repoPath);
                     }
                 }
-            } else if (!configFile.isRepoDisconnected(repoPath)) {
+            } else if (!isSyncingBranch) {
```

Comment:
Should `!configFile.isRepoDisconnected(repoPath)` be replaced with `configFile.isRepoActive(repoPath)`?

Reply:
No, using `! isSyncingBranch` is a better option here.